src/pages/ReimbursementAccount/VerifiedBankAccountFlowEntryPoint.tsx

```diff
@@ -0,0 +1,313 @@
+    toggleValidateCodeActionModal?: (isVisible: boolean) => void;
+
+    /** Set step for non USD flow */
+    setNonUSDBankAccountStep: (shouldShowContinueSetupButton: string | null) => void;
+
+    /** Set step for USD flow */
+    setUSDBankAccountStep: (shouldShowContinueSetupButton: string | null) => void;
+};
+
+const bankInfoStepKeys = INPUT_IDS.BANK_INFO_STEP;
+
+function VerifiedBankAccountFlowEntryPoint({
+    policyName = '',
+    policyID = '',
+    onBackButtonPress,
+    reimbursementAccount,
+    onContinuePress,
+    shouldShowContinueSetupButton,
+    hasForeignCurrency,
+    isValidateCodeActionModalVisible,
+    toggleValidateCodeActionModal,
+    setNonUSDBankAccountStep,
+    setUSDBankAccountStep,
+}: VerifiedBankAccountFlowEntryPointProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const [isPlaidDisabled] = useOnyx(ONYXKEYS.IS_PLAID_DISABLED);
+    const [bankAccountList] = useOnyx(ONYXKEYS.BANK_ACCOUNT_LIST);
+    const errors = reimbursementAccount?.errors ?? {};
+    const pendingAction = reimbursementAccount?.pendingAction ?? null;
+    const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
+    const optionPressed = useRef('');
+
+    const contactMethod = account?.primaryLogin ?? '';
+    const loginData = useMemo(() => loginList?.[contactMethod], [loginList, contactMethod]);
+    const validateLoginError = getEarliestErrorField(loginData, 'validateLogin');
+    const hasMagicCodeBeenSent = !!loginData?.validateCodeSent;
+    const plaidDesktopMessage = getPlaidDesktopMessage();
+    const bankAccountRoute = `${ROUTES.BANK_ACCOUNT_WITH_STEP_TO_OPEN.getRoute(policyID, REIMBURSEMENT_ACCOUNT_ROUTE_NAMES.NEW, ROUTES.WORKSPACE_INITIAL.getRoute(policyID))}`;
+    const personalBankAccounts = bankAccountList ? Object.keys(bankAccountList).filter((key) => bankAccountList[key].accountType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT) : [];
+
+    const removeExistingBankAccountDetails = () => {
+        const bankAccountData: Partial<ReimbursementAccountForm> = {
+            [bankInfoStepKeys.ROUTING_NUMBER]: '',
+            [bankInfoStepKeys.ACCOUNT_NUMBER]: '',
+            [bankInfoStepKeys.PLAID_MASK]: '',
+            [bankInfoStepKeys.IS_SAVINGS]: undefined,
+            [bankInfoStepKeys.BANK_NAME]: '',
+            [bankInfoStepKeys.PLAID_ACCOUNT_ID]: '',
+            [bankInfoStepKeys.PLAID_ACCESS_TOKEN]: '',
+        };
+        updateReimbursementAccountDraft(bankAccountData);
+    };
+
+    /**
+     * optionPressed ref indicates what user selected before modal to validate account was displayed
+     * In this hook we check if account was validated and then proceed with the option user selected
+     * note: non USD accounts only have manual option available
+     */
+    useEffect(() => {
+        if (!account?.validated) {
+            return;
+        }
+
```

Comment:
We need to hide the modal here to avoid issue https://github.com/Expensify/App/issues/73690